Mathlib/Data/Matroid/Rank/ENat.lean

```diff
@@ -298,6 +261,170 @@ lemma eRk_inter_add_eRk_union_le (M : Matroid α) (X Y : Set α) :
+
+lemma eRank_eq_top_iff (M : Matroid α) : M.eRank = ⊤ ↔ M.RankInfinite := by
+  rw [← not_rankFinite_iff, ← eRank_ne_top_iff, not_not]
+
+@[deprecated (since := "2025-04-13")] alias rankInfinite_iff_eRk_eq_top := eRank_eq_top_iff
+
+@[simp]
+lemma eRank_eq_top [RankInfinite M] : M.eRank = ⊤ :=
+  (eRank_eq_top_iff _).2 <| by assumption
+
+@[simp]
+lemma eRk_eq_top_iff : M.eRk X = ⊤ ↔ ¬ M.IsRkFinite X := by
+  obtain ⟨I, hI⟩ := M.exists_isBasis' X
+  rw [hI.eRk_eq_encard, encard_eq_top_iff, ← hI.finite_iff_isRkFinite, Set.Infinite]
+
+lemma eRk_ne_top_iff : M.eRk X ≠ ⊤ ↔ M.IsRkFinite X := by
+  simp
+
+@[simp]
+lemma eRk_lt_top_iff : M.eRk X < ⊤ ↔ M.IsRkFinite X := by
+  rw [lt_top_iff_ne_top, eRk_ne_top_iff]
+
+lemma IsRkFinite.eRk_lt_top (h : M.IsRkFinite X) : M.eRk X < ⊤ :=
+  eRk_lt_top_iff.2 h
+
+@[deprecated (since := "2025-04-13")] alias eRk_lt_top_of_finite := IsRkFinite.eRk_lt_top
+
+lemma IsRkFinite.eRk_ne_top (h : M.IsRkFinite X) : M.eRk X ≠ ⊤ :=
+  h.eRk_lt_top.ne
+
+/-! ### Constructions -/
+
+@[simp]
+lemma eRank_map {β : Type*} (M : Matroid α) (f : α → β) (hf : InjOn f M.E) :
+    (M.map f hf).eRank = M.eRank := by
+  obtain ⟨B, hB⟩ := M.exists_isBase
+  rw [← (hB.map hf).encard_eq_eRank, ← hB.encard_eq_eRank, (hf.mono hB.subset_ground).encard_image]
+
+@[simp]
+lemma eRk_map_eq {β : Type*} {f : α → β} (M : Matroid α) (hf : InjOn f M.E)
+    (hX : X ⊆ M.E := by aesop_mat) : (M.map f hf).eRk (f '' X) = M.eRk X := by
+  obtain ⟨I, hI⟩ := M.exists_isBasis X
+  rw [hI.eRk_eq_encard, (hI.map hf).eRk_eq_encard, (hf.mono hI.indep.subset_ground).encard_image]
+
+@[simp]
+lemma eRk_comap_eq {β : Type*} {f : α → β} (M : Matroid β) (X : Set α) :
+    (M.comap f).eRk X = M.eRk (f '' X) := by
+  obtain ⟨I, hI⟩ := (M.comap f).exists_isBasis' X
+  obtain ⟨hI', hinj, -⟩ := comap_isBasis'_iff.1 hI
+  rw [← hI.encard_eq_eRk, ← hI'.encard_eq_eRk, hinj.encard_image]
+
+@[simp]
+lemma loopyOn_eRk_eq (X Y : Set α) : (loopyOn Y).eRk X = 0 := by
+  obtain ⟨I, hI⟩ := (loopyOn Y).exists_isBasis' X
+  rw [hI.eRk_eq_encard, loopyOn_indep_iff.1 hI.indep, encard_empty]
+
+@[simp]
+lemma loopyOn_eRank_eq (X : Set α) : (loopyOn X).eRank = 0 := by
+  rw [eRank_def, loopyOn_eRk_eq]
+
+lemma eRank_eq_zero_iff : M.eRank = 0 ↔ M = loopyOn M.E := by
+  refine ⟨fun h ↦ closure_empty_eq_ground_iff.1 ?_, fun h ↦ by rw [h, loopyOn_eRank_eq]⟩
+  obtain ⟨B, hB⟩ := M.exists_isBase
+  rw [← hB.encard_eq_eRank, encard_eq_zero] at h
+  rw [← h, hB.closure_eq]
+
+lemma exists_of_eRank_eq_zero (h : M.eRank = 0) : ∃ X, M = loopyOn X :=
+  ⟨M.E, by simpa [eRank_eq_zero_iff] using h⟩
+
+@[simp] lemma eRank_loopyOn_eq_zero (α : Type*) : (emptyOn α).eRank = 0 := by
+  rw [eRank_eq_zero_iff, emptyOn_ground, loopyOn_empty]
+
+lemma eq_loopyOn_iff_eRank : M = loopyOn X ↔ M.eRank = 0 ∧ M.E = X :=
+  ⟨fun h ↦ by rw [h]; simp, fun ⟨h,h'⟩ ↦ by rw [← h', ← eRank_eq_zero_iff, h]⟩
+
+@[simp] lemma freeOn_eRank_eq (X : Set α) : (freeOn X).eRank = X.encard := by
```

Comment:
For the ones of this form, I think the `_eq` isn't really adding much semantic value, except in the `eq_zero_iff` lemmas, in which I agree it should be there.